### PR TITLE
[WIP] Update map matching to filter empty waypoints

### PR DESCRIPTION
**Pull request by danswick**
Addresses part of https://github.com/mapbox/mapbox-sdk-js/issues/289. 

While verifying the map-matching service, I noticed we weren't filtering empty waypoints, which would result in a `'Waypoints must be integer indexes that reference a coordinate'` error message from the Map Matching API. This PR adds a filter to make sure we don't get any pesky stray `;` in our `waypoints` list. 

@kepta for review, please 🙏 

todo: 

- [x] Update tests

### CODE REVIEW

**Comment by danswick** [comment]
Looks like we may also need to verify similar behavior for the following, though I'm not sure yet if the Map Matching API will be cranky about `;;` in the same way it is for `waypoints`.  

- [ ] approaches
- [ ] radiuses
- [ ] timestampes
- [ ] waypoint names

File: services/map-matching.js [hunk 1]
```diff
@@ -133,6 +133,9 @@ MapMatching.getMatch = function(config) {
       .map(function(val, i) {
         return val === true ? i : '';
       })
+      .filter(function(x) {
+        return x === 0 || Boolean(x);
```
Comment (kepta) [inline services/map-matching.js]: What exactly do we want to filter out in this @danswick ?

**Comment by davidtheclark** [comment]
I think the key misunderstanding here is that there's not supposed to an equal number of waypoints and coordinates — right?

**Comment by danswick** [comment]
@davidtheclark @kepta I believe there _can_ be an equal number of waypoints and coordinates, but the issue arises when they are not the same and the SDK joins lonely semicolons into the list of waypoints like: `0;;2;3`, in which case the Map Matching API will throw the error mentioned in the OP. So we need to filter out any `path` items that say `isWaypoint: false` (and also avoid accidentally filtering out the `0` index).

File: docs/services.md [hunk 2]
```diff
@@ -377,7 +377,7 @@ SDK returned.
   - `config.position` **(`"auto"` \| [Object][150])** If `"auto"`, the viewport will fit the
       bounds of the overlay(s). Otherwise, the maps' position is described by an object
       with the following properties:
-      `coordinates` (required): [`coordinates`][144] for the center of image.
+      `coordinates` (required): [`coordinates`][106] for the center of image.
```
Comment (colleenmcginnis) [inline docs/services.md]: Why is this changing?

Comment (danswick) [reply docs/services.md]: good catch! This PR got old enough that I needed to catch up to master and update the docs.

**Pull request merged**